### PR TITLE
Made random data generation generic.

### PR DESCRIPTION
**Pull request by cbsmith**
Per [discussion](https://github.com/lemire/FastPFor/pull/101#discussion_r1108965556), making this a separate pull request.

### CODE REVIEW

**Comment by lemire** [comment]
@cbsmith Thanks.

**Comment by cbsmith** [comment]
Hrm... not sure why the build failures are occurring. I'll take a look at this later tonight.

File: unittest/test_fastpfor.cpp [hunk 1]
```diff
@@ -110,26 +110,14 @@ namespace FastPForLib {
           }
         }
 
-        void _genDataRandom(std::vector<int32_t>& v, uint32_t values) {
+	template<typename T>
+        void _genDataRandom(std::vector<T>& v, uint32_t values) {
           v.clear();
           std::mt19937_64 e2(123456);
```
Comment (pps83) [inline unittest/test_fastpfor.cpp:116]: Not related to the PR, but the test might have discovered some hidden issues if it didn't use fixed seed. Perhaps, using device_random or time to seed generator and then, saving the seed is a better approach: If a test fails, then it should report the seed to be able to repro the test case.

Comment (cbsmith) [reply unittest/test_fastpfor.cpp:116]: That seems like a good suggestion.

File: unittest/test_fastpfor.cpp [hunk 2]
```diff
@@ -110,26 +110,14 @@ namespace FastPForLib {
           }
         }
 
-        void _genDataRandom(std::vector<int32_t>& v, uint32_t values) {
+	template<typename T>
+        void _genDataRandom(std::vector<T>& v, uint32_t values) {
           v.clear();
```
Comment (pps83) [inline unittest/test_fastpfor.cpp:115]:
```suggestion
        void _genDataRandom(std::vector<T>& v, size_t values) {
          v.resize(values);
```

**Review by pps83 — CHANGES_REQUESTED** [review]

File: unittest/test_fastpfor.cpp (hunk 2 above)
Comment (pps83) [reply unittest/test_fastpfor.cpp:115]: this should fix the crash

Comment (cbsmith) [reply unittest/test_fastpfor.cpp:115]: Yeah, I realized this just now as I got a chance to look at this again. I'm kind of surprised the tests passed on my own machine though. A little freaky.

File: unittest/test_fastpfor.cpp (hunk 1 above)
Comment (lemire) [reply unittest/test_fastpfor.cpp:116]: I agree that it is most helpful to use fixed seeds.

**Comment by cbsmith** [comment]
Fixed the vector sizing error. Tests should pass now.

**Comment by lemire** [comment]
Running tests now.

**Comment by lemire** [comment]
Merging.

**Pull request merged**